pkg/imageregistry/imageregistry.go

```diff
@@ -0,0 +1,328 @@
+package imageregistry
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/apiservers"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/clusteroperator"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	"github.com/openshift-kni/eco-goinfra/pkg/pod"
+	imageregistryv1 "github.com/openshift/api/imageregistry/v1"
+	v1 "github.com/openshift/api/operator/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+var imageRegistryObjName string = "cluster"
+var imageRegistryNamespace string = "openshift-image-registry"
+var imageRegistryCoName string = "image-registry"
+
+// Builder provides a struct for imageRegistry object from the cluster and a imageRegistry definition.
+type Builder struct {
+	// imageRegistry definition, used to create the imageRegistry object.
+	Definition *imageregistryv1.Config
+	// Created imageRegistry object.
+	Object *imageregistryv1.Config
+	// api client to interact with the cluster.
+	apiClient *clients.Settings
```

Comment:
Please add:

// errorMsg is processed before imageregistry Config object is created.
errorMsg string